internal/nodes/avi_vrf_translator.go

```diff
@@ -16,19 +16,74 @@ package nodes
 
 import (
 	"errors"
-	"math"
 	"regexp"
+	"sort"
 	"strconv"
 	"strings"
 
 	"github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/internal/lib"
-
+	"github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/internal/objects"
 	"github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/pkg/utils"
 
 	"github.com/vmware/alb-sdk/go/models"
 	v1 "k8s.io/api/core/v1"
 )
 
+func GetStaticRoutesForOtherNodes(aviVrfNode *AviVrfNode, routeId string) []*models.StaticRoute {
+	var staticRouteCopy []*models.StaticRoute
+	nodePrefix := lib.GetClusterName() + "-" + routeId
+	for i := 0; i < len(aviVrfNode.StaticRoutes); i++ {
+		if routeId == "" || !strings.HasPrefix(*aviVrfNode.StaticRoutes[i].RouteID, nodePrefix) {
+			staticRouteCopy = append(staticRouteCopy, aviVrfNode.StaticRoutes[i])
+		}
+	}
+	return staticRouteCopy
+}
+
+func (o *AviObjectGraph) CheckAndDeduplicateRecords(key string) {
+	aviVrfNodes := o.GetAviVRF()
+	if len(aviVrfNodes) == 0 {
+		return
+	}
+	// Each AKO should have single VRF node as it deals with single cluster only.
+	aviVrfNode := aviVrfNodes[0]
+
+	podCidrNextHopMap := make(map[string]string)
+	hasDuplicateRecords := false
+	// Check if duplicate records for staticroutes exist
+	for i := 0; i < len(aviVrfNode.StaticRoutes); i++ {
+		_, ok := podCidrNextHopMap[*aviVrfNode.StaticRoutes[i].Prefix.IPAddr.Addr]
+		if ok {
+			hasDuplicateRecords = true
```

Comment:
We can have a log here mentioning duplicate records are found for the pod cidr

Reply:
Done